From: Andrew Morton <akpm@linux-foundation.org>
Subject: selftests-mm-add-merge-test-for-partial-msealed-range-fix
Date: Mon Mar 30 05:12:35 PM PDT 2026

fix build

In file included from merge.c:4:
merge.c: In function merge_merge_vmas_with_mseal:
merge.c:1229:27: error: __NR_mseal undeclared (first use in this function)
 1229 |         ASSERT_EQ(syscall(__NR_mseal, 0, 0, ~0UL), -1);
      |                           ^~~~~~~~~~

Cc: Lorenzo Stoakes (Oracle) <ljs@kernel.org>
Cc: David Hildenbrand <david@kernel.org>
Cc: Jann Horn <jannh@google.com>
Cc: Liam Howlett <liam.howlett@oracle.com>
Cc: Michal Hocko <mhocko@suse.com>
Cc: Mike Rapoport <rppt@kernel.org>
Cc: Pedro Falcato <pfalcato@suse.de>
Cc: Shuah Khan <shuah@kernel.org>
Cc: Suren Baghdasaryan <surenb@google.com>
Cc: Vlastimil Babka <vbabka@kernel.org>
Signed-off-by: Andrew Morton <akpm@linux-foundation.org>
---

 tools/testing/selftests/mm/merge.c |    1 +
 1 file changed, 1 insertion(+)

--- a/tools/testing/selftests/mm/merge.c~selftests-mm-add-merge-test-for-partial-msealed-range-fix
+++ a/tools/testing/selftests/mm/merge.c
@@ -6,6 +6,7 @@
 #include <fcntl.h>
 #include <stdio.h>
 #include <stdlib.h>
+#include <asm-generic/unistd.h>
 #include <unistd.h>
 #include <sys/mman.h>
 #include <sys/prctl.h>
_
